pinot-core/src/main/java/com/linkedin/pinot/core/periodictask/PeriodicTaskScheduler.java

```diff
@@ -70,11 +73,19 @@ public void start(List<PeriodicTask> periodicTasks) {
     }
   }
 
+  /**
+   * Shutdown executor service and stop the periodic tasks
+   */
```

Comment:
Do we need synchronization between start and stop? What if the leadership change happens quickly and start is still exeucting, when stop kicks in?

Reply:
@sunithabeeram yes that's a good point, added the synchronization between start and stop